sklearn/cross_validation.py

```diff
@@ -595,15 +595,50 @@ def __len__(self):
 class BaseShuffleSplit(with_metaclass(ABCMeta)):
     """Base class for ShuffleSplit and StratifiedShuffleSplit"""
 
-    def __init__(self, n, n_iter=10, test_size=0.1, train_size=None,
-                 random_state=None):
-        self.n = n
-        self.n_iter = n_iter
-        self.test_size = test_size
-        self.train_size = train_size
-        self.random_state = random_state
-        self.n_train, self.n_test = _validate_shuffle_split(n, test_size,
-                                                            train_size)
+    def __init__(self, *args, **kwargs):
```

Comment:
My intention to refactor like this is to distinguish whether the None is specified by user
or just didn't pass a value to it:

e.g.
by using "ShuffleSplit(10)" we shall init by default (.9/.1 split)
but
by using "ShuffleSplit(10, None, None)" we shall raise an exception 

Using parameter list like (self, n, n_iter=10) we couldn't achieve it.
Using parameter list like (*args, **kwargs) we can achieve this by:

if len(args) < 3 and "test_size" in kwargs:
    # the test_size=None is generated by using 
    # "ShuffleSplit(10)", user didn't pass a value
    # to parameter test_size and train_size
else:
    # the test_size=None is generated by using 
    # "ShuffleSplit(10, test_size=None, train_size=None)"
    # we shall raise ValueError()

Is there any better opinions for achieving this? I shall support that.